smart-py/objkt_swap.py

```diff
@@ -17,135 +38,679 @@ def __init__(self, objkt, hdao, manager, metadata, curate):
             genesis = sp.timestamp(0),
             curate = curate,
             locked = False
-            )
-    
+        )
+
     @sp.entry_point
     def genesis(self):
-        sp.verify((sp.sender == self.data.manager) & ~(self.data.locked))
+        sp.verify(
+            (
+                sp.sender == self.data.manager
+            ) & ~(
+                self.data.locked
+            )
+        )
+
+        # what's up with adding 45 days?
         self.data.genesis = (sp.now).add_days(45)
         self.data.locked = True
-    
+
     @sp.entry_point
     def update_manager(self, params):
-        sp.verify(sp.sender == self.data.manager)
+        sp.verify(
+            sp.sender == self.data.manager
+        )
+
         self.data.manager = params
-        
+
     @sp.entry_point
     def swap(self, params):
-        sp.verify((params.objkt_amount > 0))
-        self.fa2_transfer(self.data.objkt, sp.sender, sp.to_address(sp.self), params.objkt_id, params.objkt_amount)
-        self.data.swaps[self.data.swap_id] = sp.record(issuer=sp.sender, objkt_id=params.objkt_id, objkt_amount=params.objkt_amount, xtz_per_objkt=params.xtz_per_objkt)
+        # must be at least one edition
+        sp.verify(params.objkt_amount > 0)
+        # no more than 10k
+        sp.verify(params.objkt_amount <= 10000)
+        # the objkt id must be at least 1
+        # TODO make this at least the starting number (152 atm)
+        sp.verify(params.objkt_id > 0)
+        # the objkt number should not be beyond the max number
+        sp.verify(params.objkt_id <= self.data.objkt_id)
+
+        # get the objkt
+        # TODO how to get the details of the objkt???
+        c = sp.contract(
```

Comment:
This doesn't seem right, according to the docs of the sc.contract method:
https://smartpy.io/reference.html#_contracts_and_addresses

Reply:
this was where i got stuck and the last line i was on before i pushed :)